### PR TITLE
Closes xyzz/acquisition#167

### PR DESCRIPTION
**Pull request by GrappigPanda**
Calls new method AutoOnline::SetOnlineUpdate on QCloseEvent.

### CODE REVIEW

**Comment by xyzz** [comment]
I think it would be better if AutoOnline's SendOnlineUpdate had a `bool online` as its only argument. This way you don't have to construct the URL outside of the class.

**Comment by GrappigPanda** [comment]
That's a better way of doing it and it gives me an excuse to remove GetUrl() which leaves a foul taste in my mouth. I'll push in a sec.

**Comment by GrappigPanda** [comment]
Okay, bro. updated.

File: src/autoonline.cpp [hunk 1]
```diff
@@ -91,20 +91,25 @@ static bool IsPoeRunning() {
...
+
+void AutoOnline::Check() {
+    bool running = IsPoeRunning();
 
     if (running || previous_status_) {
-        QNetworkRequest request(QUrl(url.c_str()));
-        QByteArray data;
-        request.setHeader(QNetworkRequest::UserAgentHeader, (std::string("Acquisition ") + VERSION_NAME).c_str());
-        nm_.post(request, data);
+        SendOnlineUpdate(true);
```
Comment (xyzz) [inline src/autoonline.cpp]: Should be SendOnlineUpdate(running)

**Comment by GrappigPanda** [comment]
Not necessarily. running can be false, but if previous_status_ is true, the if statement will still execute SendOnlineUpdate.

**Comment by xyzz** [comment]
Yes but if `running` is false it should set the status to "offline" status, not "online".

**Comment by GrappigPanda** [comment]
Then should it just be?
<code>

void AutoOnline::Check() {

    bool running = IsPoeRunning();

    SendOnelineUpdate(running);
  
    previous_status_ = running;

    emit Update(running);
}
</code>

**Comment by xyzz** [comment]
Nope, it should be just like it is now but with `SendOnlineUpdate(running)` instead of `SendOnlineUpdate(true)`. 

The reason there's this `if` is done: we update the status either if the user is still online and running is true (and the 1 hour status is prolonged) or if the user went offline after being online and the previous_status is true while running is false (so that the system marks him or her as offline).

The only time the status isn't updated is when the user is offline and the previous status is offline too, which means either the user wasn't online to begin with or that we refreshed the status the last time, this is done to not create additional load on poe.trade servers.

**Comment by GrappigPanda** [comment]
Ohhh, okay. Sorry about all the trouble haha.

**Comment by GrappigPanda** [comment]
Let's hope this one's the one.

**Pull request merged**